File: app/src/main/java/com/maple/audiometry/base/BaseFragment.java
package com.maple.audiometry.base;

import android.content.Context;
import android.os.Bundle;
import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentManager;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

/**
 * @author maple
 * @time 16/4/13
 */
public abstract class BaseFragment extends Fragment {
    public View view;
    public Context mContext;
    public FragmentManager fm;

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        this.mContext = getActivity();
        this.fm = getFragmentManager();
    }


    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        view = initView(inflater);
        view.setClickable(true);
        return view;
    }


    @Override
    public void onActivityCreated(Bundle savedInstanceState) {
        initData(savedInstanceState);
        initListener();
        super.onActivityCreated(savedInstanceState);
    }


    public abstract View initView(LayoutInflater inflater);


    public abstract void initData(Bundle savedInstanceState);


    public abstract void initListener();


}

File: app/src/main/java/com/maple/audiometry/ui/base/BaseFragment.kt
package com.maple.audiometry.ui.base

import android.content.Context
import androidx.fragment.app.Fragment

/**
 * Fragment基类
 *
 * @author maple
 * @time 2018/12/22
 */
abstract class BaseFragment : Fragment() {
    lateinit var mContext: Context

    override fun onAttach(context: Context) {
        super.onAttach(context)
        mContext = context
    }

    open fun onKeyBackPressed(): Boolean {
        // 是否消耗掉back事件
        return false
    }
}

